### DISCUSSION TITLE
DVR and multiple Edge nodes

### DISCUSSION THREAD

**Discussion by hernanrz** [OvenMediaLabs/OvenMediaEngine#1425] [Help]
We have LLHLS with DVR enabled on two Edge nodes, clients connect to one of the two nodes randomly, we're seeing a situation where if a client connects to edge2 (for example) and edge2 hasn't pulled the OVT stream from the origin before (for instance, no clients had connected to this node in a while) they get a shorter DVR playlist than the one available in edge1. Is there a way to ensure both edge nodes share the same dvr playlist? Or, make an edge instantly pull a stream when it starts, so they both have a full DVR available.

Some ideas I've had regarding this:

- Use NFS to share the dvr folder between edge nodes
- Periodically run rsync to sync the folders
- Writing an application to make edge pull streams from origin when they start, to create a full DVR

**Comment by getroot** [comment]
I actually didn't assume that the Edge would perform DVR or Dump functions. My assumption is that it should be done on LLHLS Origin and Edge are different cache servers. So I didn't think about inter-Edge synchronization.

I still don't quite understand what the benefit would be of configuring a pipeline like yours (OME Origin - OME Edge - Nginx).

**Reply by naanlizard** [reply]
Origin and edge separate even on one server so that we can adjust edge settings without disrupting incoming streams. Nginx because connecting directly to ome edge is unacceptably high CPU usage

Not to mention we want to scale edge servers eventually so practicing on a single server is good
